src/fcmp_pp/fcmp_pp_types.h

```diff
@@ -67,20 +68,78 @@ struct HeliosT final
 using OutputBytes = ::OutputBytes;
 using OutputChunk = ::OutputSlice;
 //----------------------------------------------------------------------------------------------------------------------
+// Define C++ type here so it can be used in FFI types
+using FcmpPpProof = std::vector<uint8_t>;
+//----------------------------------------------------------------------------------------------------------------------
+//----------------------------------------------------------------------------------------------------------------------
+// FFI types
+//----------------------------------------------------------------------------------------------------------------------
+// FFI types instantiated on the Rust side must be destroyed back on the Rust side. We wrap them in a unique ptr with a
+// custom deleter that calls the respective Rust destroy fn.
+#define DEFINE_FCMP_FFI_TYPE(raw_t, cpp_fn)                                      \
+    struct raw_t##Deleter { void operator()(raw_t##Unsafe *p) const noexcept; }; \
+    using raw_t = std::unique_ptr<raw_t##Unsafe, raw_t##Deleter>;                \
+    raw_t cpp_fn;
+
+// Sometimes a shared ptr is necessary so that we can reference the same copy of underlying data in multiple places.
+#define DEFINE_FCMP_FFI_SHARED_TYPE(raw_t, cpp_fn1, cpp_fn2) \
+    using raw_t = std::shared_ptr<raw_t##Unsafe>;            \
```

Comment:
```suggestion
    using raw_t##Shared = std::shared_ptr<raw_t##Unsafe>;            \
```

I'd like the `shared_ptr` name to be differentiable from the `unique_ptr` name. Also allows us to have unique *and* shared variants for the same raw pointer type.